src/test/java/org/dependencytrack/resources/v1/FindingResourceTest.java

```diff
@@ -589,6 +596,225 @@ public void getAllFindingsGroupedByVulnerabilityWithAclEnabled() {
+        Vulnerability v2 = createVulnerability("Vuln-2", Severity.HIGH, "Vuln Title 2", "   Yet another description but with surrounding whitespaces   ", "", 46);
+        Vulnerability v3 = createVulnerability("Vuln-3", Severity.LOW, "Vuln Title 3", "A description-with-hyphens-(and parentheses)", "  Recommendation with whitespaces  ", 23);
+
+        // Note: Same vulnerability added to multiple components to test whether "rules" field doesn't contain duplicates
+        qm.addVulnerability(v1, c1, AnalyzerIdentity.NONE);
+        qm.addVulnerability(v2, c1, AnalyzerIdentity.NONE);
+        qm.addVulnerability(v3, c1, AnalyzerIdentity.NONE);
+        qm.addVulnerability(v3, c2, AnalyzerIdentity.NONE);
+
+        Response response = target(V1_FINDING + "/project/" + project.getUuid().toString()).request()
+            .header(HttpHeaders.ACCEPT, MEDIA_TYPE_SARIF_JSON)
+            .header(X_API_KEY, apiKey)
+            .get(Response.class);
+
+        Assert.assertEquals(200, response.getStatus(), 0);
+        Assert.assertEquals(MEDIA_TYPE_SARIF_JSON, response.getHeaderString(HttpHeaders.CONTENT_TYPE));
+        final String jsonResponse = getPlainTextBody(response);
+
+        assertThatJson(jsonResponse)
+            .withMatcher("version", equalTo(new About().getVersion()))
+            .withMatcher("fullName", equalTo("OWASP Dependency-Track - " + new About().getVersion()))
+            .withMatcher("vuln1Id", equalTo(v1.getVulnId()))
+            .withMatcher("vuln2Id", equalTo(v2.getVulnId()))
+            .withMatcher("vuln3Id", equalTo(v3.getVulnId()))
+            .withMatcher("vuln1CweId", equalTo(v1.getCwes().get(0).toString()))
+            .withMatcher("vuln2CweId", equalTo(v2.getCwes().get(0).toString()))
+            .withMatcher("vuln3CweId", equalTo(v3.getCwes().get(0).toString()))
+            .withMatcher("vuln1Desc", equalTo(v1.getDescription().trim()))
+            .withMatcher("vuln2Desc", equalTo(v2.getDescription().trim()))
+            .withMatcher("vuln3Desc", equalTo(v3.getDescription().trim()))
+            .withMatcher("vuln1Level", equalTo(getSARIFLevelFromSeverity(v1.getSeverity())))
+            .withMatcher("vuln2Level", equalTo(getSARIFLevelFromSeverity(v2.getSeverity())))
+            .withMatcher("vuln3Level", equalTo(getSARIFLevelFromSeverity(v3.getSeverity())))
+            .withMatcher("vuln3Recommendation", equalTo(v3.getRecommendation().trim()))
+            .withMatcher("comp1Purl", equalTo(c1.getPurl().toString()))
+            .withMatcher("comp2Purl", equalTo(c2.getPurl().toString()))
+            .withMatcher("comp1Name", equalTo(c1.getName()))
+            .withMatcher("comp2Name", equalTo(c2.getName()))
+            .withMatcher("comp1Group", equalTo(c1.getGroup()))
+            .withMatcher("comp2Group", equalTo(c2.getGroup()))
+            .withMatcher("comp1Version", equalTo(c1.getVersion()))
+            .withMatcher("comp2Version", equalTo(c2.getVersion()))
```

Comment:
For these you don't need matchers, you can use literals in the JSON instead. Only use matchers for values that you have no control over (e.g. auto-generated UUIDs, `About` values).